docker-compose.yml

```diff
@@ -0,0 +1,113 @@
+version: '3.9'
+
+services:
+
+  traefik:
+    image: traefik:v2.8
+    restart: always
+    container_name: "${DOCKER_NAME}_traefik"
+    command:
+      - --providers.docker=true
+      # Enable the API handler in insecure mode,
+      # which means that the Traefik API will be available directly
+      # on the entry point named traefik.
+      - --api.insecure=true
+      # Defines the path to the configuration file with the certificates list.
+      - --providers.file.filename=/root/.config/ssl.toml
+      # Define Traefik entry points to port [80] for http and port [443] for https.
+      - --entrypoints.web.address=:80
+      - --entrypoints.websecure.address=:443
+    networks:
+      # Define the network on which traefik is going to operate.
+      - web
+    ports:
+      # Open traefik http [80] and https [443] ports.
+      - '80:80'
+      - '443:443'
+    volumes:
+      - ./traefik/dynamic_conf.yaml:/etc/traefik/dynamic_conf.yaml:ro
+      - ./traefik/traefik.yaml:/etc/traefik/traefik.yaml:ro
+      - ./certs:/etc/certs:ro
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.${DOCKER_NAME}_traefik.rule=Host(`traefik.${DOCKER_BASE_URL}`)"
+      - "traefik.http.routers.${DOCKER_NAME}_traefik.tls=true"
+      # # Enable Traefik API handler entrypoint on http.
+      - "traefik.http.routers.${DOCKER_NAME}_traefik-http.entrypoints=web"
+      # # By default the Traefik API handler operates on the port [8080].
+      # # Define a load balancer to route the entry point to [8080].
+      - "traefik.http.services.${DOCKER_NAME}_traefik.loadbalancer.server.port=8080"
+
+  terrareg:
+    container_name: "${DOCKER_NAME}_app"
+    restart: unless-stopped
+    build: .
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.${DOCKER_NAME}_app.tls=true"
+      - "traefik.http.routers.${DOCKER_NAME}_app.rule=Host(`terrareg.${DOCKER_BASE_URL}`)"
+      - "traefik.http.services.${DOCKER_NAME}_app.loadbalancer.server.port=5000"
+      - "traefik.http.routers.${DOCKER_NAME}_app.service=${DOCKER_NAME}_app"
+    depends_on:
+      - mysql
+    # environment:
+    #   - SAML2_PRIVATE_KEY
+    #   - SAML2_PUBLIC_KEY
+    env_file:
+      - ./.env
+    volumes:
+      - terrareg:/app/data
+      - ./keys:/app/keys
+      - ~/.ssh/:/root/.ssh/
+    networks:
+      web:
+
+  mysql:
+    container_name: "${DOCKER_NAME}_mysql"
+    restart: unless-stopped
+    image: docker.io/bitnami/mysql:8.0
```

Comment:
Is there any reason for using bitnami, as opposed to the official MariaDB/MySQL images?

Reply:
I've run into challenges passing in configuration changes to the official containers in the past so I tend to default to the bitnami images. Since there's no custom config needed for the mysql conf file in this case this can be changed to the official images if you prefer.